app/assets/javascripts/components/samples.jsx

```diff
@@ -148,6 +156,113 @@ class Samples extends React.Component {
     })
   }
 
+  //Load more samples on scroll
+  scrollDown() {
+    var that = this;
+    $(window).scroll(function() {
+      if ($(window).scrollTop() > $(document).height() - $(window).height() - 60) {
+        { !that.state.isRequesting && !that.state.pageEnd ? that.loadMore() : null }
+        return false;
+      }
+    });
+  }
+
+  //fetch first set of samples
+  fetchSamples() {
+    const params = this.getParams();
+    axios.get(`/samples${params}`).then((res) => {
+      if (!res.data.samples.length) {
+        this.setState({ displayEmpty: true });
+      }
+      this.setState((prevState) => ({
+        loading: false,
+        initialFetchedSamples: res.data.samples,
+        allSamples: res.data.samples,
+        displayEmpty: false,
+        pagesLoaded: prevState.pagesLoaded+1,
+        totalNumber: res.data.total_count
+      }))
+    }).catch((err) => {
+      this.setState((prevState) => ({
+        loading: false,
+        allSamples: [],
+        pagesLoaded: 0,
+      }))
+    })
+  }
+
+  //fetch all Projects
+  fetchProjects() {
+    axios.get(`/projects.json`).then((res) => {
+      this.setState({ 
+        allProjects: res.data,
+        loading: false,
+      })
+    }).catch((err) => {
+      this.setState({ 
+        allProjects: [],
+        loading: false,
+      })
+    })
+  }
+
+  //fetch data used by projects page
+  fetchProjectPageData() {
+    this.setState({ loading: true })
+    this.fetchProjects();
+    this.fetchSamples();
+  }
+
+  //load more paginated samples
+  loadMore() {
+    const params = this.getParams();
+    this.setState({ isRequesting: true })
+    axios.get(`/samples${params}`).then((res) => {
+      this.setState((prevState) => ({
+        isRequesting: false,
+        allSamples: [...prevState.allSamples, ...res.data.samples],
+        pagesLoaded: prevState.pagesLoaded+1,
+        pageEnd: res.data.samples.length >= 15 ? false : true,
+      }))
+    }).catch((err) => {
+      this.setState((prevState) => ({
+        isRequesting: false,
+        allSamples: [...prevState.allSamples],
+        pagesLoaded: prevState.pagesLoaded,
+        pageEnd: prevState.pageEnd
+      }))
+    })
+  }
+
+  //fetch project, filter and search params
+  getParams() {
+    const projectId = parseInt(this.state.selectedProjectId);
+    
+    const paramsWithProject = `?filter=${this.state.filterParams}&page=${this.state.pagesLoaded+1}&project_id=${projectId}&search=${this.state.searchParams}`;
```

Comment:
simplify this with 
let params = `?filter=${this.state.filterParams}&page=${this.state.pagesLoaded+1}&search=${this.state.searchParams}`
if project_id !== null {
  params += `&project_id=${project_id}` 
}
return params